lib/modules/editor/widgets/cell_tile.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:andax/models/cell.dart';
+import 'package:andax/models/translation_asset.dart';
+import 'package:andax/modules/editor/screens/story.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class CellTile extends StatelessWidget {
+  const CellTile(
+    this.cell, {
+    this.onTap,
+    Key? key,
+  }) : super(key: key);
+
+  final Cell? cell;
+  final VoidCallback? onTap;
+
+  @override
+  Widget build(BuildContext context) {
+    return ListTile(
+      leading: cell == null
+          ? null
+          : Icon(
+              cell!.display == null
+                  ? Icons.visibility_off_rounded
+                  : cell!.display == CellDisplay.check
+                      ? Icons.check_circle_rounded
+                      : cell!.display == CellDisplay.text
+                          ? Icons.short_text_rounded
+                          : Icons.linear_scale_rounded,
+            ),
```

Comment:
This is extremely difficult to read. I suggest adding the following map to the class:
```dart
static const Map<CellDisplay?, IconData> _displayIcon = {
  null:              Icons.visibility_off_rounded,
  CellDisplay.check: Icons.check_circle_rounded,
  CellDisplay.text:  Icons.short_text_rounded,
  CellDisplay.range: Icons.linear_scale_rounded,
};
```
and then replacing all of this with a simple
```dart
leading: Icon(_displayIcon[cell?.display]),
```

Reply:
I also wanted to point out that my solution has `Icons.visibility_off_rounded` as the default (if `cell?.display` was null), whereas yours had it as `Icons.linear_scale_rounded`. I'm not sure which one is the intended (I assume the former), so I wanted to clarify.